From: Ian Jackson <ian.jackson@citrix.com>
Date: Wed, 3 Oct 2018 16:25:58 +0100
X-Dgit-Generated: 4.17.2+76-ge1f9cb16e2-1 259ad0bb8cf8a8ef22f32be3164537553080971d
Subject: autoconf: Provide libexec_libdir_suffix

This is going to be used to put libfsimage.so into a path containing
the multiarch triplet.

Signed-off-by: Ian Jackson <ian.jackson@citrix.com>

---

diff --git a/m4/paths.m4 b/m4/paths.m4
index e4104bcce0fb..1093c7cdf9f5 100644
--- a/m4/paths.m4
+++ b/m4/paths.m4
@@ -86,6 +86,13 @@ XEN_SCRIPT_DIR=$xen_scriptdir_path
 AC_SUBST(XEN_SCRIPT_DIR)
 AC_DEFINE_UNQUOTED([XEN_SCRIPT_DIR], ["$XEN_SCRIPT_DIR"], [Xen's script dir])
 
+dnl Allows .../libexec/lib (used for libfsimage) to have a multiarch path
+AC_ARG_WITH([libexec-libdir-suffix],
+    AS_HELP_STRING([--with-libexec-libdir-suffix=/SUFFIX],
+    [Name of subdirectory or suffix to use after ...LIBEXEC/lib.]),
+    [libexec_libdir_suffix=$withval],
+    [libexec_libdir_suffix=''])
+
 AC_ARG_WITH([xen-dumpdir],
     AS_HELP_STRING([--with-xen-dumpdir=DIR],
     [Path to directory for domU crash dumps. [LOCALSTATEDIR/lib/xen/dump]]),
@@ -120,7 +127,7 @@ dnl These variables will be substituted in various .in files
 LIBEXEC_BIN=${LIBEXEC}/bin
 AC_SUBST(LIBEXEC_BIN)
 AC_DEFINE_UNQUOTED([LIBEXEC_BIN], ["$LIBEXEC_BIN"], [Xen's libexec path])
-LIBEXEC_LIB=${LIBEXEC}/lib
+LIBEXEC_LIB=${LIBEXEC}/lib${libexec_libdir_suffix}
 AC_SUBST(LIBEXEC_LIB)
 LIBEXEC_INC=${LIBEXEC}/include
 AC_SUBST(LIBEXEC_INC)
